src/inmanta/deploy/tasks.py

```diff
@@ -136,6 +136,7 @@ async def execute(self, task_manager: "scheduler.TaskManager", agent: str, reaso
         # Full status of the deploy,
         # may be unset if we fail before signaling start to the server, will be set if we signaled start
         deploy_result: DeployResult | None = None
+        deploy_state: state.DeploymentResult = state.DeploymentResult.FAILED
```

Comment:
I have a strong preference for not setting a value here, and to require each branch to set it to the appropriate value. i.e. anywhere `success` is currently set.

Additionally, `deploy_state` is a strange name, since it's a deployment result. If we want to avoid ambiguity with `deploy_result`, I'd rather rename that one with more context, e.g. `executor_deploy_result`. Then, if you still find it ambiguous, name this one `scheduler_deployment_result`.